trainee_template/ablation.amlg

```diff
@@ -0,0 +1,363 @@
+            )
+        ))
+        
+        (if 
+            ;if context_values are not null, we are reacting to a new case
+            (not (= context_values (null)))
+            (accum "react_kwargs" (assoc
+                context_values context_values
+            ))
+            ;elif case_id is not null, we are reacting to an existing case
+            (not (= case_id (null)))
+            (accum "react_kwargs" (assoc
+                case_indices
+                    (retrieve_from_entity case_id (list ".session" ".session_training_index") )
+                preserve_feature_values context_features
+                leave_case_out (true)
+            ))
+        )
+
+        ;compute the entropy of the influence weights as retrieved by react
+        (entropy (map
+            (lambda (get (current_value) ".influence_weight") )
+
+            (get (call React react_kwargs) "influential_cases")
+        ))
+    )
+
+    ;used to compute and store the influence weight entropies for call cases contained in the Trainee in #Analyze
+    ; context_features: list of context features to use when determining influential cases.
+    ; label_name: optional name of the feature to store influence weight entropies in.
+    #ComputeAndStoreInfluenceWeightEntropies
+    (let 
+        (assoc
+            influence_weight_entropy_map
+                ||(map
+                    (lambda
+                        (call ComputeInfluenceWeightEntropy (assoc
+                            case_id (current_index 1)
+                            context_features context_features
+                            use_case_weights use_case_weights
+                            weight_feature weight_feature
+                        ))
+                    )
+                    (zip (call AllCases))
+                )
+        )
+
+        (call StoreCaseValues (assoc
+            case_values_map influence_weight_entropy_map
+            label_name
+                (if
+                    (= (null) label_name)
+                    internalLabelInfluenceWeightEntropy
+                    label_name
+                )
+        ))
+
+        (accum_to_entities (assoc revision 1))
+        (assign_to_entities (assoc hasInfluenceWeightEntropies (true)))
+    )
+
+    ;initialize parameters and internal features related to auto ablation
+    ; weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+    #InitializeAutoAblate
+    (declare
+        (assoc
+            weight_feature autoAblateWeightFeature
+        )
+
+        (seq
+            (assign_to_entities (assoc hasPopulatedCaseWeight (true)))
+            (accum_to_entities (assoc revision 1))
+            (call CreateCaseWeights (assoc
+                feature_name weight_feature
+            ))
+        )
+        
+    )
+
+    ;determine whether a new case (one that is not in the Trainee) should be ablated (trained as weights) or kept.
```

Comment:
Maybe I'm just a fool, but I would appreciate a blip in the method description that specifies if a return value of 1 indicates the case should be ablated or not.